src/url-state-machine.js

```diff
@@ -960,6 +960,8 @@ URLStateMachine.prototype["parse path start"] = function parsePathStart(c) {
     if (c !== p("/")) {
       --this.pointer;
     }
+  } else if (this.stateOverride && this.url.host == null) {
```

Comment:
```suggestion
  } else if (this.stateOverride && this.url.host === null) {
```
Since the rest of the code uses strict equality comparison, should we do the same here too?